streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java

```diff
@@ -2330,4 +2333,101 @@ public void shouldNotAllowNullNamedOnProcessWithStores() {
         assertThat(exception.getMessage(), equalTo("named can't be null"));
     }
 
+
+    @Test
+    public void shouldKTableFromKStream() {
+        final Consumed<String, String> consumed = Consumed.with(Serdes.String(), Serdes.String())
+            .withName("table-process");
+        final Consumed<String, String> consumed2 = Consumed.with(Serdes.String(), Serdes.String())
+            .withName("stream-table-process");
+
+        final StreamsBuilder builder = new StreamsBuilder();
+
+        final String topic1 = "topic1";
+
+        final KTable<String, String> table1 = builder.table(topic1, consumed);
+
+        final MockProcessorSupplier<String, Object> supplier = new MockProcessorSupplier<>();
```

Comment:
We should not use `MockProcessorSupplier` but write the result into an output topic and use a `TestOutputTopic` for verification of the result (the `MockProcessorSupplier` should be removed completely mid-term...)